src/main/java/uk/gov/hmcts/reform/professionalapi/configuration/FeatureConditionEvaluation.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.hmcts.reform.professionalapi.configuration;
+
+import static uk.gov.hmcts.reform.professionalapi.controller.constants.ProfessionalApiConstants.ERROR_MESSAGE_403_FORBIDDEN;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+@Component
+public class FeatureConditionEvaluation  implements HandlerInterceptor {
+
+    @Value("${deleteOrganisationEnabled}")
+    private boolean deleteOrganisationEnabled;
+
+    @Override
+    public boolean preHandle(HttpServletRequest request,
+                             HttpServletResponse response, Object arg2) throws Exception {
+
+        if (!deleteOrganisationEnabled && request.getMethod().equals("DELETE")) {
+
```

Comment:
you can avoid the  if condition  if you do interceptor with API endpoint as you wont have to check for request method in that case

Reply:
ok